### DISCUSSION TITLE
Getting "No files available for downloading" when selecting multiple files to download

### DISCUSSION THREAD

**Discussion by andrewlow** [photoprism/photoprism#2396] [General]
I went digging in the code - and got here https://github.com/photoprism/photoprism/blob/182bc09d87b86310b076e44a3b6302898cff4efc/internal/api/download_zip.go#L76

Using Firefox 100. 

If I select only 1 item - it downloads fine. 
If I select 2 items - I get this error. Fails for 3 photos as well.

Feels like a bug - but not sure where to start debugging

**Comment by andrewlow** [comment]
Also fails in Chrome Version 102.0.5005.61 (Official Build) (x86_64)

Select 2 photos - try to download.. boom

**Reply by andrewlow** [reply]
Running in a container - pulled the latest update last night.

**Reply by andrewlow** [reply]
super weird - selecting 2 video files results in a download just fine.

2 photos + 1 video - download is fine

Remove the 1 video from above - and just have the same 2 photos -- and download is broken

**Comment by andrewlow** [comment]
ahh. but the zip file does not contain the photos.. just the video(s)

**Comment by andrewlow** [comment]
Hmm.. using the demo app - selecting the 

- Bench https://demo.photoprism.app/browse#&gid=1&pid=6
- Peacock https://demo.photoprism.app/browse#&gid=1&pid=7

and downloading - works fine.

**Reply by graciousgrey** [reply]
What type of photos (RAW, JPG...) are you trying to download ? 
And what download settings are defined in your settings.yml ?

The default download settings for downloads from the context menu are:
Originals: true
MediaRaw: false
MediaSidecar: false

Originals:true means only files from your originals folder will be downloaded, not from sidecar.

Downloading from the files tab of the edit dialog is not affected by these settings.

**Reply by andrewlow** [reply]
JPG files - sourced from my Pixel 4a phone.

The issue here is that if I only select 1 of the photos - the download works.  This indicates it is not a setting issue.
Selecting multiple files - does not work.

From my settings.yml

```
  Download: true
```

I have no settings in settings.yml for any of: Originals, MediaRaw, MediaSidecar - this may be due to my running an old photoprism and having upgraded it by pulling newer containers over time - but I assume default values would still apply in my case then.

**Reply by andrewlow** [reply]
Oh boo -- I setup the development version of photoprism and tested.. it's fine for multi-file downloads.

https://docs.photoprism.app/developer-guide/setup/

I'm going to investigate if this is a problem with the setting.yml .. maybe some default isn't what it should be?

**Reply by andrewlow** [reply]
Well - it looks like my settings.yml file was missing some of the Download fields.

What I had..
```
Download:
  Name: file
```

This was an artifact of my install being from quite some time ago - and I've just pulled newer containers to update. This _should_ work IMO - but ok.

When I modified the settings.yml file to have
```
Download:
  Name: file
  Disabled: false
  Originals: true
  MediaRaw: false
  MediaSidecar: false
  ```
I was able to multi-select files - and download a zip just fine.

**Reply by andrewlow** [reply]
Seems odd - all I did was give it the defaults in theory -- but the behaviour changed? 

https://github.com/photoprism/photoprism/blob/182bc09d87b86310b076e44a3b6302898cff4efc/internal/config/settings_download.go#L5

**Comment by andrewlow** [comment]
Grr.. I thought I'd fixed this with the config change - new photos uploaded - I can download in a zip. Older ones (from a different phone) cannot be downloaded. I've enabled debug, but there is only this in the logs.

```
time="2022-06-10T15:14:06Z" level=debug msg="api: abort /api/v1/zip with code 404 (No files available for download)"
time="2022-06-10T15:14:06Z" level=debug msg="http: POST /api/v1/zip (404) [15.109375ms]"
```

**Reply by andrewlow** [reply]
Hmm. it seems I can't download photos that were uploaded prior to my changing the config?

**Reply by andrewlow** [reply]
Nope.. not quite. 

It seems off by a day? I can download photos from the day before I fixed things - but older ones I cannot?! So confusing.

Net - I still have normal .jpg files I cannot download in a zip. If I select only 1 photo at a time I can download just fine.

**Comment by andrewlow** [comment]
I have created an issue - and will continue tracking things there. https://github.com/photoprism/photoprism/issues/2412

**Reply by andrewlow** [reply]
With some help from @lastzero I was able to resolve my problem. Details in issue #2412 

Short version
- Make sure you have all of the values for Download set in `settings.yml` (might not be required)
- Perform a re-scan of your library

Consider this question 'answered'
